sdk/state/close.go

```diff
@@ -0,0 +1,148 @@
+package state
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/stellar/experimental-payment-channels/sdk/txbuild"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+)
+
+// The steps for a channel coordinated close are as followed:
+// 1. Initiator or Responder submits latest declaration tx
+// (in steps 2-4 Initiator and Responder are interchangeable, as long as they alternate)
+// 2. Initiator calls ProposeCoordinatedClose
+// 3. Responder calls ConfirmCoordinatedClose
+// 4. Initiator calls ConfirmCoordinatedClose
+
+type CoordinatedClose struct {
+	observationPeriodTime      time.Duration
+	observationPeriodLedgerGap int64
+	closeSignatures            []xdr.DecoratedSignature
+}
+
+func (cc CoordinatedClose) CloseSignatures() []xdr.DecoratedSignature {
+	return cc.closeSignatures
+}
+
+func (c *Channel) CloseTxs() (*txnbuild.Transaction, *txnbuild.Transaction, error) {
+	txDecl, err := txbuild.Declaration(txbuild.DeclarationParams{
+		InitiatorEscrow:         c.initiatorEscrowAccount().Address,
+		StartSequence:           c.startingSequence,
+		IterationNumber:         c.latestCloseAgreement.IterationNumber,
+		IterationNumberExecuted: 0,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	txClose, err := txbuild.Close(txbuild.CloseParams{
+		ObservationPeriodTime:      c.observationPeriodTime,
+		ObservationPeriodLedgerGap: c.observationPeriodLedgerGap,
+		InitiatorSigner:            c.initiatorSigner(),
+		ResponderSigner:            c.responderSigner(),
+		InitiatorEscrow:            c.initiatorEscrowAccount().Address,
+		ResponderEscrow:            c.responderEscrowAccount().Address,
+		StartSequence:              c.startingSequence,
+		IterationNumber:            c.latestCloseAgreement.IterationNumber,
+		AmountToInitiator:          c.initiatorClaimAmount(),
```

Comment:
ya  you're right there will most likely need to be locking. I think it would be best to add that when we start thinking about networking, since that's when it will most be needed.

I added a ticket to track that thought: https://github.com/stellar/experimental-payment-channels/issues/84